src/libs/TaskUtils.ts

```diff
@@ -0,0 +1,51 @@
+import Onyx, {OnyxEntry} from 'react-native-onyx';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {Report} from '@src/types/onyx';
+import ReportAction from '@src/types/onyx/ReportAction';
+import * as CollectionUtils from './CollectionUtils';
+import * as Localize from './Localize';
+
+const allReports: Record<string, Report> = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT,
+    callback: (report, key) => {
+        if (!key || !report) {
+            return;
+        }
+        const reportID = CollectionUtils.extractCollectionItemID(key);
+        allReports[reportID] = report;
+    },
+});
```

Comment:
This basically line-to-line duplicate of [this block](https://github.com/Expensify/App/blob/c7eb0e96d1f87012ac4a9578d360240809b91732/src/libs/ReportActionsUtils.ts#L23-L34), and maybe others.

Reply:
@tgolen what would you say? Can we reuse utils inside other utils instead of using `Onyx.connect`?

Reply:
That is a very nuanced topic, so I apologize in advance for the long comment, but I want you to hear my thinking.

1. There is no technical problem with having the same `connect()` in two different libs. The only "problem" is that it is repetitive code, which is a somewhat valid argument. However, let's take that logic all the way to the grand conclusion. If we only intend to have one `connect()` for every key, then that basically results in some kind of file that contains all the `connect()` calls for every key. This then creates an odd in-memory type store of everything that Onyx already has in memory. This is just a strange place to end up in.

2. Having duplicate `connect()`s in the files that need them allows them to be custom-tailored for fast lookups or custom maps depending on what data is being accessed. For example, rather than doing a `.filter()` or `.map()` every time the data needs accessed, the `connect()` callback should store the data in an efficient way so that it can be easily and quickly accessed like `reports[reportID]` and there is no need to filter or map. This allows for code to be much more optimized, rather than kept in a very general and slow method (because it has to account for different libs wanting to access it in different ways).